osu.Game/Rulesets/RulesetInfo.cs

```diff
@@ -44,10 +44,13 @@ public bool Equals(RulesetInfo? other)
             if (ReferenceEquals(this, other)) return true;
             if (other == null) return false;
 
+            if (OnlineID >= 0 && other.OnlineID >= 0)
+                return OnlineID == other.OnlineID;
```

Comment:
We don't do this in other models – why are we doing it here?

Reply:
I don't think we should be doing this..

Reply:
Now that I look at how things are structured for `APIBeatmap`, I think `APIRuleset` is what we want going forward, with a custom `Equals` implementation that uses `MatchesOnlineID`. Curious if that's something you're against?

Reply:
That's fine, I just don't want one `Equals` implementation of models to not match the others (right now, every implementation is the same across ruleset/beatmap/skin/etc).